pyof/foundation/network_types.py

```diff
@@ -147,12 +145,11 @@ class TLVWithSubType(GenericTLV):
     field and a new "*sub_value*" field.
     """
 
-    def __init__(self, tlv_type=1, sub_type=7, sub_value=None):
+    def __init__(self, tlv_type=1, sub_type=7, sub_value=BinaryData()):
         """Create an instance and set its attributes."""
-        super().__init__(tlv_type=tlv_type)
+        super().__init__(value=None)
```

Comment:
@macartur just one more thing, you can pass the "tlv_type" on the `super().__init__()`, together with the value, since the parent class already set the tlv_type.
```
def __init__(self, tlv_type=1, sub_type=7, sub_value=BinaryData()):
    ......
    super().__init__(tlv_type, value=None)
    super.sub_type = sub_type
    super.sub_value = sub_value
```